demo/src/main/java/com/example/demo/repository/CaseFileDao.java

```diff
@@ -0,0 +1,70 @@
+package com.example.demo.repository;
+
+import com.example.demo.model.CaseFile;
+import com.example.demo.model.DTO.CaseFileDTO;
+import com.example.demo.model.DTO.EntityDTOConverter;
+import com.example.demo.model.Person;
+import org.springframework.stereotype.Component;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class CaseFileDao {
+
+    @PersistenceContext
+    private final EntityManager entityManager;
+
+    public CaseFileDao(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public Optional<CaseFileDTO> create(CaseFileDTO caseFileDTO) {
+        if (caseFileDTO == null) {
+            return Optional.empty();
+        }
+        Person person;
+        Long personId = caseFileDTO.getPersonId();
+
+        if (personId != null) {
+            person = entityManager.find(Person.class, personId);
+        } else {
+            person = EntityDTOConverter.convertToPersonEntity(caseFileDTO.getPersonDTO());
+            entityManager.persist(person);
+        }
+        CaseFile caseFile = EntityDTOConverter.convertToCaseFileEntity(caseFileDTO);
+        caseFile.setPerson(person);
+        entityManager.persist(caseFile);
+
+        caseFileDTO.setId(caseFile.getId());
+        return Optional.of(caseFileDTO);
+    }
+
+    public List<CaseFile> listAllCaseFiles() {
+        return entityManager.createNamedQuery("CaseFile.findAll", CaseFile.class).getResultList();
+    }
+
+    public CaseFile getCaseFileById(Long id) {
+        return entityManager.createNamedQuery("CaseFile.findById", CaseFile.class)
+                .setParameter("id", id)
+                .getSingleResult();
+    }
```

Comment:
Consider method type migration from CaseFile to: **Optional<CaseFile>** . 

ex:

```
Query query = entityManager.createNamedQuery("CaseFile.findById", CaseFile.class)
                .setParameter("id", id);

try 
{
return Optional.of(query.getSingleResult());
} catch(NoResultException e)
{
return Optional.empty();
}
```